src/Mvc/src/Microsoft.AspNetCore.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcCoreBuilderExtensions.cs

```diff
@@ -166,8 +166,9 @@ internal static void AddViewServices(IServiceCollection services)
             services.TryAddTransient<IHtmlHelper, HtmlHelper>();
             services.TryAddTransient(typeof(IHtmlHelper<>), typeof(HtmlHelper<>));
             services.TryAddSingleton<IHtmlGenerator, DefaultHtmlGenerator>();
-            services.TryAddSingleton<ExpressionTextCache>();
-            services.TryAddSingleton<IModelExpressionProvider, ModelExpressionProvider>();
+            services.TryAddSingleton<ModelExpressionProvider>();
```

Comment:
What do we need from the implementation that prevents this being registered as `IModelExpressionProvider`?

Reply:
:ok:

Thinking out loud: Should we do the get-then-return-instance trick to also register this singleton as `IModelExpressionProvider`? If this is the only one we recommend users deal with, there's no need. But, if you see a scenario where the `interface` is what we'd recommend, go ahead and add that registration.